pkg/internal/approver/validation/validator.go

```diff
@@ -0,0 +1,87 @@
+package validation
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/ext"
+)
+
+const (
+	varSelf    = "self"
+	varRequest = "cr"
+)
+
+// Validator knows how to validate CSR attribute values in CertificateRequests
+// against CEL expressions declared in CertificateRequestPolicy.
+// A Validator must be compiled before it can validate. A compiled Validator
+// is stateless, thread-safe, and cacheable.
+type Validator interface {
+	// Compile compiles the Validator CEL expression into a runnable program
+	// that can perform validations. Any compilation errors will be returned.
+	Compile() error
+
+	// Validate validates the supplied value against the Validator CEL
+	// expression in context of the request.
+	// Returns 'true' if the value is valid (passes validation).
+	// A returned error should be considered as an internal/technical error,
+	// and not typically user-facing.
+	Validate(value string, request cmapi.CertificateRequest) (bool, error)
+}
+
+type validator struct {
+	expression string
+	program    cel.Program
+}
+
+func (v *validator) Compile() error {
+	if v.program != nil {
+		// Already compiled
+		return nil
+	}
+
+	env, err := cel.NewEnv(
+		cel.Variable(varSelf, cel.StringType),
+		cel.Variable(varRequest, cel.MapType(cel.StringType, cel.StringType)),
```

Comment:
^^ I think you already said that we could do this in a future PR?

Reply:
Believe me, I tried to get the protobuf-thing to work, but no luck so far. No matter how I did it, the CEL compiler didn't manage to find my type. And the CEL documentation is not awsome. If you can help me, I would appreciate it. It would be optimal to have the `cr` variable of type `metav1.ObjectMeta` - and the protos are already there. I just cannot make them work in CEL! 😡

Reply:
I already spent significant time trying to get the protos to work, so if I cannot get any help, I suggest staying with Map for now and improve it in a follow-up PR. Others might have more experience with protos than me.

Reply:
I think I finally made it work - at least locally on my workstation. 😊 CI seems to struggle finding the correct release of `protoc` to download. Any pointers are appreciated @inteon. See https://github.com/cert-manager/approver-policy/pull/277/commits/046b3624e8cb1ee2fc8e9e4d8f672ef58827a821.

Reply:
By replacing `uname -p` with `uname -m` in the detect-OS Makefile, I was able to get a step further. Now the problem is no `unzip` in the container? that runs CI. This should be possible to fix, but how? I am far outside my comfort zone here. Please help! 🙏

Reply:
This should be resolved now: using a protobuf message as variable in expressions.